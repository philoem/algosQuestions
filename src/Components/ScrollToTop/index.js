import React, { useEffect, useState } from 'react'
import { FaArrowAltCircleUp } from 'react-icons/fa'
import styles from './index.module.css'

const ScrollToTop = () => {
	const [visible, setVisible] = useState(false)

	const toggleVisible = () => {
		const scrolled = document.documentElement.scrollTop
		if (scrolled > 100) {
			setVisible(true)
		} else if (scrolled <= 100) {
			setVisible(false)
		}
	}

	const scrollingToTop = () => {
		window.scrollTo({
			top: 0,
			behavior: 'smooth'
		})
	}

	useEffect(() => {
		window.addEventListener('scroll', toggleVisible)
	}, [])

	return (
		<>
			<button onClick={scrollingToTop} className={styles.buttonToTop}>
				{visible && <FaArrowAltCircleUp />}
			</button>
		</>
	)
}

export default ScrollToTop
