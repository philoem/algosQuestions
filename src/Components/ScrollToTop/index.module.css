.buttonToTop {
  position: fixed;
  left: 95%;
  bottom: 40px;
  height: 20px;
  font-size: 3rem;
  z-index: 1;
  cursor: pointer;
  color: #173570;
  border: none;
  background-color: rgba(255, 255, 255, 0);
}