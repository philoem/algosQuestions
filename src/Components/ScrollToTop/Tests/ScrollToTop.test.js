import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import { defineFeature, loadFeature } from 'jest-cucumber'
import ScrollToTop from '../../ScrollToTop/index'
import userEvent from '@testing-library/user-event'

const feature = loadFeature('src/Components/ScrollToTop/Tests/ScrollToTop.feature')

defineFeature(feature, (test) => {
	let container

	const givenScrollToTopIsRendered = (given, comp) => {
		given('ScrollToTop component is rendered', () => {
			container = render(comp).container
		})
	}

	const andRequiredElementsAreMounted = (and) => {
		and('Required elements are mounted', () => {
			expect(screen.getByRole('button')).toBeInTheDocument()
		})
	}

	const andDefaultStyleIsApplied = (and) => {
		and('Default style is applied', () => {
			expect(container).toMatchSnapshot()
		})
	}

	test('ScrollToTop is display', ({ given, when, then, and }) => {
		givenScrollToTopIsRendered(given, <ScrollToTop />)
		andRequiredElementsAreMounted(and)
		andDefaultStyleIsApplied(and)

		when('User scroll to end of list', () => {
			fireEvent.scroll(window, { target: { scrollY: 100 } })
		})

		then("ScrollToTop's icon is display", () => {
			const button = screen.getByRole('button')
			expect(button).toBeEnabled()
		})

		and("ScrollToTop's button is clicked", () => {
			userEvent.click(screen.getByRole('button'))
		})
	})
})
