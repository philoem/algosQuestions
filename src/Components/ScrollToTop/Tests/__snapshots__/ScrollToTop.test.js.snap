// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollToTop ScrollToTop is display 1`] = `
<div>
  <button
    class="buttonToTop"
    type="button"
  />
</div>
`;
